app/Http/Controllers/PostController.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\DB;
+use App\Models\Tag;
+use App\Models\Post;
+
+class PostController extends Controller
+{
+    public function create(Request $request) {
+        $requestTags = $request->get('tags');
+        $abc = Tag::whereIn('name', $requestTags)->pluck('name')->toArray();
+        $newTags = array_diff($requestTags, $abc);
+        foreach ($newTags as $tag) {
+            DB::table('tags')->insert([
+                'name' => $tag
+            ]);
+        }
+        $newPost = Post::create([
+            'title' => $request->get('title'),
+            'user_id' => $request->get('user_id'),
+            'content' => $request->get('content'),
+            'status' => 'pending',
+        ]);
+
+        $allTags = DB::table('tags')->get();
+        foreach ($allTags as $tag) {
+            DB::table('post_tag')->insert([
+                'post_id' => $newPost->id,
+                'tag_id' => $tag->id,
+            ]);
```

Comment:
tham khảo cách sử dụng detach. Theo mình hiểu là đoạn này mỗi `post` tạo mới đều có toàn bộ `tags` trong DB ???